lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -51,30 +53,42 @@ data NetworkLayer t m = NetworkLayer
         -- ^ Broadcast a transaction to the chain producer
     }
 
--- | Network is not reachable
-newtype ErrNetworkUnreachable
+-- | Network is unavailable
+data ErrNetworkUnavailable
     = ErrNetworkUnreachable Text
+      -- ^ Cannot connect to network backend.
+    | ErrNetworkInvalid Text
+      -- ^ Network backend reports that the requested network is invalid.
     deriving (Generic, Show, Eq)
 
-instance Exception ErrNetworkUnreachable
+instance Exception ErrNetworkUnavailable where
+    displayException = \case
+        ErrNetworkUnreachable err -> T.unpack err
+        ErrNetworkInvalid netName ->
+            "Invalid network name: " ++ T.unpack netName
+
+-- | Exception predicate for 'ErrNetworkUnreachable'.
+isNetworkUnreachable :: ErrNetworkUnavailable -> Bool
+isNetworkUnreachable (ErrNetworkUnreachable _) = True
+isNetworkUnreachable _ = False
```

Comment:
As outlined here: https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#avoid-wildcard-when-pattern-matching-on-adts

This is a typical case where using a wildcard could cause issues in the future. I think we do want the compiler to warn us about a non-handled pattern match case if we were to add a new branch to `ErrNetworkUnavailable`. With a wildcard, we might conflates two distinct case under the same behavior.

```suggestion
isNetworkUnreachable :: ErrNetworkUnavailable -> Bool
isNetworkUnreachable = \case
    ErrNetworkUnreachable{} -> True
    ErrNetworkInvalid{} -> False
```

Reply:
I thought that this was the typical case where using a wildcard _would not_ cause issues in the future, but have changed it anyway...